x/ibc/light-clients/06-solomachine/types/misbehaviour_handle.go

```diff
@@ -49,16 +49,23 @@ func (cs ClientState) CheckMisbehaviourAndUpdateState(
 
 // verifySignatureAndData verifies that the currently registered public key has signed
 // over the provided data and that the data is valid. The data is valid if it can be
-// unmarshaled into the specified data type.
+// unmarshaled into the specified data type or the timestamp of the signature is less
+// than the consensus state timestamp.
 func verifySignatureAndData(cdc codec.BinaryMarshaler, clientState ClientState, misbehaviour *Misbehaviour, sigAndData *SignatureAndData) error {
+	// timestamp less than consensus state would always fail and not succeed in fooling the
+	// light client
+	if sigAndData.Timestamp < clientState.ConsensusState.Timestamp {
```

Comment:
cc @cwgoes this differs from the spec, but I think it is correct. Should we keep this check? It just stops clients from being frozen if the timestamp signed was not recent enough to fool the light client into passing verification or update checks

Reply:
Yes, that makes sense to me. Thanks for the ping. Spec changes made in https://github.com/cosmos/ics/commit/128082e0180354d7c7b7763e9063a3afa227eaf5.